### DISCUSSION TITLE
Struggling to get gamepads working

### DISCUSSION THREAD

**Discussion by alexwaterston** [excaliburjs/Excalibur#3199] [Q&A]
Hello, I'm trying to get gamepads working in my simple Game and Watch style game. I've enabled gamepad support in the game and connected a USB gamepad but none of the gamepad objects in the list is "connected". I'm on a mac and I've tried the game in both the arc browser and safari. The Jelly Jumper sample on the excalibur site doesn't work either. Anyone got any experience of this? Thank you! Alex

**Comment by eonarheim** [comment]
@alexwaterston That's definitely unexpected! Can you post some example code that you have? Could you also send me the version of Arc/Safari and your Mac OS? I should be able to partially test on browserstack... 

I do remember that Mac needs drivers for some gamepads to work, what gamepad do you have?

Jelly Jumper is working with a gamepad on my windows machine at the moment.

> The Jelly Jumper sample on the excalibur site doesn't work either. Anyone got any experience of this?

Just checking that the problem you're having with Jelly Jumper is the gamepads not working, but the game is running?

**Comment by alexwaterston** [comment]
Thanks Erik. Really appreciate you replying so quickly.

My code is here: https://github.com/alexwaterston/phonegame

And you can play the game here:
<https://alexwaterston.github.io/phonegame/>
Super Service Desk Dash<https://alexwaterston.github.io/phonegame/>
alexwaterston.github.io<https://alexwaterston.github.io/phonegame/>
[X]<https://alexwaterston.github.io/phonegame/>

I got it working in safari in the end but it still doesn’t work in arc. I’m using the latest public arc build and I am on the latest macOS.

It seems to work on windows with no issues.

Jelly jumper runs but again, in arc, controllers do not work.

Alex

Alex Waterston
Associate Director - Innovation & Leadership
(He/Him)
T: +44 345 094 0945
M: +447479690210
waterstons.com
www.waterstons.com/Alex-Waterston
Click to receive our newsletter
​Waterstons Limited. Registered in England and Wales No. 3818424
Our registered office is at Liddon Court, Aykley Heads, Durham, England, DH1 5TS
DISCLAIMER: The information contained in this email is intended for the named recipient only. It may contain confidential information. If you are not the intended recipient, you must not copy, distribute or take any action in reliance on it. Please note that neither Waterstons Limited nor the sender accepts any responsibility for viruses and it is your responsibility to scan attachments (if any). 

On 3 Sep 2024, at 18:17, Erik Onarheim ***@***.***> wrote:

﻿

@alexwaterston<https://github.com/alexwaterston> That's definitely unexpected! Can you post some example code that you have? Could you also send me the version of Arc/Safari and your Mac OS? I should be able to partially test on browserstack...

I do remember that Mac needs drivers for some gamepads to work, what gamepad do you have?

Jelly Jumper is working with a gamepad on my windows machine at the moment.

The Jelly Jumper sample on the excalibur site doesn't work either. Anyone got any experience of this?

Just checking that the problem you're having with Jelly Jumper is the gamepads not working, but the game is running?
